app/controllers/homepage_controller.rb

```diff
@@ -0,0 +1,14 @@
+class HomepageController < ApplicationController
+  skip_before_action :authorize
+
+  def index
+    @q = Question.where.not(published_at: nil).order('created_at DESC')
+    if current_user
+      @q = @q.where.not(user_id: current_user.id).ransack(params[:q])
+    else
+      @q = @q.ransack(params[:q])
+    end
+
+    @questions = @q.result.includes(:user)
```

Comment:
includes method is only defined on ActiveRecord::Relation object hence it is not calleable on @q since it is object of Ransack::Search class. Therefore includes called after calling result method

Reply:
put includes above